### ISSUE TITLE
Middleware doesn't work in route group.

### ISSUE THREAD

**Issue by Emanuel-Palestino** [leafsphp/router#6]
Hello, I'm developing a Leaf API and I want to use a custom middleware on a group of routes but it doesn't work.
This is my code:
```
app()->registerMiddleware('auth', function () {
	$auth_header= Leaf\Http\Headers::get('Authorization');

	if (!$auth_header) {
		response()->exit([
			'message' => 'Autenticación requerida.'
		], 401);
	}
});

app()->group('/auth', ['middleware' => 'auth', function () {
    /* USUARIOS */
    app()->mount('/usuarios', function () {
        app()->get('/', 'UsuariosController@index');
        app()->post('/', 'UsuariosController@store');
        app()->get('/{id}', 'UsuariosController@show');
        app()->put('/{id}', 'UsuariosController@update');
        app()->delete('/{id}', 'UsuariosController@destroy');
    });

    /* other groups */
}]);
```

When I make the request, the middleware is ignored.

The middleware works when I use it on a single route.
```
app()->get('/', ['middleware' => 'auth', 'UsuariosController@index']);
```

I tried to use the docs example https://leafphp.dev/docs/routing/middleware.html#named-middleware-route-options-new , but I got an error:
```
Leaf\Router::group(): Argument #1 ($path) must be of type string, array given, ...
```

I hope you can help me, thanks.

**Comment by mychidarko** [comment]
Hi @Emanuel-Palestino, I just released a new version that fixes this issue (v0.2.3). The wrong info on the documentation has been corrected as well. You can run `leaf install router` or `composer require leafs/router` to fix the issue in your codebase.

**Comment by Emanuel-Palestino** [comment]
Thank you so much, now it works.
